src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String paragraph = System.lineSeparator();
+        StringBuilder builder = new StringBuilder("Report for period ")
+                .append(dateFrom).append(" - ")
+                .append(dateTo);
+
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate endDate = LocalDate.parse(dateTo, formatter);
+
+        for (String name: names) {
+            int totalSalary = 0;
+
+            for (String employeeData: data) {
+                String[] employeeInfo = employeeData.split(" ");
+                LocalDate workDay = LocalDate.parse(employeeInfo[0], formatter);
+
+                if (name.equals(employeeInfo[1])
+                        && !workDay.isAfter(endDate)
+                        && !workDay.isBefore(startDate)) {
+                    totalSalary += Integer.parseInt(employeeInfo[2])
+                            * Integer.parseInt(employeeInfo[3]);
+                }
+            }
```

Comment:
It's hard to understand what is stored under arrayName[someIndex] without knowing the input format, it's always better to extract such expressions in separate variables with proper names to give some context on what is actually going on